src/main/java/com/example/shorturl/dao/UserRepository.java

```diff
@@ -0,0 +1,30 @@
+package com.example.shorturl.dao;
+
+import com.example.shorturl.model.user.UserEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface UserRepository extends JpaRepository<UserEntity, Long> {
+
+    /**
+     * This method is used to look for a UserEntity with a specific
+     * userName from the database
+     *
+     * @param userName This is the username to look for
+     * @return Optional<UserEntity> This returns the found user entity
+     */
+    Optional<UserEntity> findByUserName(String userName);
+
+    /**
+     * This method is used to look for the UserEntities with a specific
+     * userName or email from the database
+     *
+     * @param userName This is the userName to look for
+     * @param email This is the email to look for
+     * @return List<UserEntity> This returns the found user entities
+     */
+    List<UserEntity> findByUserNameOrEmail(String userName, String email);
```

Comment:
But you still use the list here
Why can't you use two separate methods or just use UserRequest as a argument?

Reply:
Previously I used two separate methods, but we agreed not to use them because we were retrieving twice from the database. I can implement that again if you want. I was thinking on using a @Query and limit the result to just 1 entity